### PR TITLE
updated a couple of dependencies and added optics

### PR DESCRIPTION
**Pull request by Karnich**
added optics to the project. It posts results to https://optics.apollodata.com/

closes nothing but took a couple of screen dumps and made it easy to make more

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,19 +33,20 @@
     "@types/hapi": "13.0.37",
     "@types/koa": "2.0.33",
     "@types/mongoose": "4.5.39",
+    "@types/node-uuid": "6.0.26-alpha",
     "@types/nodemailer": "1.3.32",
-    "@types/node-uuid": "0.0.28",
-    "apollo-server": "0.3.3",
```
Comment (Crevil) [inline package.json:38]: We are not using `apollo-server` any more?

File: src/server.ts [hunk 2]
```diff
@@ -1,35 +1,48 @@
...
-import { apolloExpress, graphiqlExpress } from "apollo-server";
+import { graphqlExpress, graphiqlExpress } from "graphql-server-express";
 import { makeExecutableSchema } from "graphql-tools";
-import { json } from "body-parser";
+import * as bodyParser from "body-parser";
 import Schema from "./data/schema";
 import Resolvers from "./data/resolvers";
 import { MongooseConnection } from "./data/connectors";
 
+import OpticsAgent from "optics-agent";
```
Comment (Crevil) [inline src/server.ts:10]: Did you end up making a PR for types or are they missing for this?

File: src/server.ts [hunk 3]
```diff
@@ -1,35 +1,48 @@
...
+
 import { AlgorithmMediator } from "./algorithms";
 
 const GRAPHQL_PORT = 8080;
 
 (mongoose as any).Promise = global.Promise;
 
-const graphQLServer = express();
+const app = express();
```
Comment (Crevil) [inline src/server.ts:18]: Why the rename?

**Review by Crevil — APPROVED** [review]

**Comment by Karnich** [comment]
Im not really sure where to put the type definition >.<
About apollo server.. Apparantly the new cool is using the express graphql server or something.. thats how they do it now

**Comment by Crevil** [comment]
@Karnich In root is fine by now. When it's pushed to NPM with official typings, we can migrate to that.

Cool cool. I like the new cool stuff. ;)

**Pull request merged**